server/room.mjs

```diff
@@ -122,6 +122,16 @@ export default class Room {
         player.send(func, args);
   }
 
+  deleteUnusedVariantFiles(stateID, oldState, newState) {
+    for(const oldVariantID in oldState.variants) {
+      if(!newState.variants[oldVariantID] && stateID.match(/^[a-z0-9]+$/) && oldVariantID.match(/^[a-z0-9]+$/)) {
+        const savefile = path.resolve() + '/save/states/' + this.id + '-' + stateID + '-' + oldVariantID + '.json';
```

Comment:
With these changes, this now will be 6 places throughout this file that  construct a "savefile" name for a state or a room. Consider creating function(s) that do this to eliminate duplicate code.

Reply:
Yeah, I already introduced the function in another PR.